src/models/Package.vala

```diff
@@ -0,0 +1,165 @@
+    public string version { get; set; }
+    public string description { get; set; }
+    public Gee.List<string> keywords { get; set; }
+    public string homepage { get; set; }
+    public Bugs bugs { get; set; }
+    public string license { get; set; }
+    public Person author { get; set; }
+    public Gee.List<Person> contributors { get; set; }
+    public Gee.List<FundingInfo> funding { get; set; }
+    public Gee.List<string> files { get; set; }
+    public Repository repository { get; set; }
+    public Gee.Map<string, string> dependencies { get; set; }
+    public Gee.Map<string, string> dev_dependencies { get; set; }
+    public Gee.Map<string, string> optional_dependencies { get; set; }
+
+    public static Package from_json (Json.Node node) {
+        assert (node.get_node_type () == OBJECT);
+        return (Package) Json.gobject_deserialize (typeof (Package), node);
+    }
+
+    public Json.Node to_json () {
+        return Json.gobject_serialize (this);
+    }
+
+    private bool deserialize_property (
+        string property_name,
+        out Value @value,
+        ParamSpec pspec,
+        Json.Node property_node
+    ) {
+        switch (property_name) {
+            case "keywords":
+                if (property_node.get_node_type () != ARRAY) {
+                    @value = {};
+                    return false;
+                }
+
+                @value = string_list_from_json (property_node);
+                
+                return true;
+
+            case "bugs":
+                if (property_node.get_node_type () == OBJECT) {
+                    @value = Bugs.from_json (property_node);
+                    return true;
+                }
+
+                @value = new Bugs () {
+                    url = property_node.get_string (),
+                };
+
+                return true;
+            
+            case "author":
+                if (property_node.get_node_type () == OBJECT) {
+                    @value = Person.from_json (property_node);
+                    return true;
+                }
+
+                if (property_node.get_value_type () != Type.STRING) {
+                    return false;
+                }
+
+                var regex = /^(.*)(?:\s)(<.*>)(?:\s)(\(.*\))/;
+                MatchInfo info;
+                
+                if (regex.match (property_node.get_string (), 0, out info)) {
+                    @value = new Person () {
+                        name = info.fetch (1),
+                        email = info.fetch (2),
+                        url = info.fetch (3),
+                    };
+
+                    return true;
+                }
+
+                return false;
+
+            default:
+                return default_deserialize_property (
+                    property_name,
+                    out @value,
+                    pspec,
+                    property_node
+                );
+        }
+    }
+}
+
+class Vamp.Repository : Object {
```

Comment:
Should this (and others) not extend `Json.Serializable`, given that we are overriding the methods?

Reply:
Also, shall we keep all of these in the top-level namespace? Would it be an overkill to separate them into a `Vamp.Package` namespace?

Reply:
> Should this (and others) not extend Json.Serializable, given that we are overriding the methods?

Repository is not overwriting any method. to_json() does not belong to the interface, and from_json() is a static method: static methods cannot be inherited in any way, and it does not belong to JsonSerializable either.

Both are methods that I simply implemented for convenience when working with them.

Reply:
> Also, shall we keep all of these in the top-level namespace? Would it be an overkill to separate them into a Vamp.Package namespace?

Well, at this point I wouldn't know much what to answer. My way of working with Vala is to just have 1 namespace and that's it, mostly because my code usually tries to be compatible with C and GObject Introspection and:
1. In C, it is not comfortable to have long namespaces, especially because you will have to write them all the time (see `gtk_window_present()`, `soup_session_send_and_read_async()`, etc).
2. GObject Introspection does not support nested namespaces, so if you create them in Vala, for Python, JS, Rust, and other languages they will become part of the class name (for example, `Vamp.Package.Repository` in Python becomes `Vamp.PackageRepository`).
3. Many Vala projects usually ask you not to use the `using` keyword, but to be explicit, so we are back to the annoyances of point 1.

In fact, I was considering dropping namespaces entirely and only using them if we are going to expose a library, and even then, using them only in classes/methods that will be exposed as public API.

Reply:
Although my position on this is not definitive, I just stated the way I work. If you really believe that we should take advantage of the use of namespaces (although I really don't know what real advantage they have in Vala), then I will use them.

Reply:
Fair enough. I suggested it just for cognitive ease, but if it makes accessing more complex, let's just stick to this. I don't see us exposing a library any time soon, and if we do, that's a problem for another day.